Carrito.js

```diff
@@ -21,20 +24,18 @@ class Carrito {
   }
 
   calcularTotal() {
-    let total = 0;
-    for (const item of this.productos) {
-      total += item.subtotal;
-    }
-    return total;
+    return this.productos.reduce((acc, p) => acc + p.subtotal, 0);
   }
 
-  mostrarResumen() {
-    let mensaje = "🛒 RESUMEN DEL PEDIDO 🛒\n\n";
+  obtenerResumenHTML() {
```

Comment:
Estas mezclando interacción con lógica de negocio, todo lo que es eventos y DOM debe estar en el main principal, la clase solo resolver lógica de negocio

Reply:
En vez de un mostrar resumenHTML deberías devolver un objeto estructurado JS y que luego el main lo renderiza como lo requiera:
```suggestion
obtenerResumen() {
  return {
    productos: this.productos.map(p => ({
      nombre: p.nombre,
      cantidad: p.cantidad,
      subtotal: p.subtotal
    })),
    total: this.calcularTotal()
  };
}
```